src/lib/routes/controller.ts

```diff
@@ -64,6 +65,16 @@ const checkPrivateProjectPermissions = () => async (req, res, next) => {
     return res.status(404).end();
 };
 
+const checkOpenAPIValidationError = () => async (err, req, res, next) => {
+    if (err?.status && err.validationErrors) {
+        const apiError = fromOpenApiValidationErrors(req, err.validationErrors);
+
+        res.status(apiError.statusCode).json(apiError);
+    } else {
+        next(err);
+    }
+};
```

Comment:
Do you see a problem on having this as a const? It's the same function all the time, we don't have to create new functions for each option path
```suggestion
const checkOpenAPIValidationError = async (err, req, res, next) => {
    if (err?.status && err.validationErrors) {
        const apiError = fromOpenApiValidationErrors(req, err.validationErrors);

        res.status(apiError.statusCode).json(apiError);
    } else {
        next(err);
    }
};
```

Reply:
@gastonfournier 

I adjust your suggestion on [b4702e2](https://github.com/Unleash/unleash/pull/8913/commits/b4702e2eeff113e548ab27837a553034b768696a) commit.

I just followed the way of `checkPrivateProjectPermissions` function definition like the one below but I agreed with your suggestion.

```TypeScript
const checkPrivateProjectPermissions = () => async (req, res, next) => {
    if (
        !req.checkPrivateProjectPermissions ||
        (await req.checkPrivateProjectPermissions())
    ) {
        return next();
    }
    return res.status(404).end();
};
```